pycobertura/reporters.py

```diff
@@ -234,6 +244,38 @@ def generate(self):
         return tabulate(lines, headers=headers)
 
 
+class CsvReporterDelta(DeltaReporter):
+    not_available = "-"
+
+    def generate(self, delimiter):
+        lines = self.get_report_lines()
+
+        list_of_lines = [headers_without_missing]
+        list_of_lines.extend(
+            [[f"{item}" for item in row[:-1]] for row in zip(*lines.values())]
+        )
+
+        if self.show_source:
+            list_of_lines = [headers_with_missing]
+            list_of_lines.extend(
+                [[f"{item}" for item in row[:-1]] for row in zip(*lines.values())]
+            )
+            for count in range(1, len(list_of_lines)):
+                missing_line = lines["Missing"][count - 1]
+                list_of_lines[count] += [
+                    f"{[self.color_number(number) for number in missing_line]}".encode(
+                        "utf-8"
+                    ).decode("unicode_escape")
+                ]
+
+        if "\\n" in repr(delimiter):
+            delimiter = "\n"
+        if "\\t" in repr(delimiter):
+            delimiter = "\t"
```

Comment:
no, the condition is mutually exclusive

Reply:
Isn't if-else mutually exclusive?

```python
if "\\n" in repr(delimiter):
    delimiter = "\n"
elif "\\t" in repr(delimiter):
    delimiter = "\t"
```

Otherwise, the delimiter will be parsed a second time unnecessarily for `\t` even if `\n` was already detected.

Reply:
resolved with https://github.com/aconrad/pycobertura/pull/130/commits/7c7635fc525d5d531a4826b415bda76d8bba0eac